sdk/appconfiguration/Azure.ResourceManager.AppConfiguration/src/Custom/AppConfigurationKeyValueCollection.cs

```diff
@@ -0,0 +1,92 @@
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        [Obsolete("This method is obsolete as it never works, it will be removed in a future release", false)]
+        public virtual AsyncPageable<AppConfigurationKeyValueResource> GetAllAsync(string skipToken = null, CancellationToken cancellationToken = default)
+        {
+            throw null;
+        }
+
+        /// <summary>
+        /// Lists the key-values for a given configuration store.
+        /// <list type="bullet">
+        /// <item>
+        /// <term>Request Path</term>
+        /// <description>/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppConfiguration/configurationStores/{configStoreName}/keyValues</description>
+        /// </item>
+        /// <item>
+        /// <term>Operation Id</term>
+        /// <description>KeyValues_ListByConfigurationStore</description>
+        /// </item>
+        /// <item>
+        /// <term>Default Api Version</term>
+        /// <description>2022-05-01</description>
+        /// </item>
+        /// <item>
+        /// <term>Resource</term>
+        /// <description><see cref="AppConfigurationKeyValueResource"/></description>
+        /// </item>
+        /// </list>
+        /// </summary>
+        /// <param name="skipToken"> A skip token is used to continue retrieving items after an operation returns a partial result. If a previous response contains a nextLink element, the value of the nextLink element will include a skipToken parameter that specifies a starting point to use for subsequent calls. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns> A collection of <see cref="AppConfigurationKeyValueResource"/> that may take multiple service requests to iterate over. </returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        [Obsolete("This method is obsolete as it never works, it will be removed in a future release", false)]
+        public virtual Pageable<AppConfigurationKeyValueResource> GetAll(string skipToken = null, CancellationToken cancellationToken = default)
+        {
+            throw null;
+        }
+
+        IEnumerator<AppConfigurationKeyValueResource> IEnumerable<AppConfigurationKeyValueResource>.GetEnumerator()
+        {
+            yield break;
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            yield break;
+        }
+
+        [Obsolete("This method is obsolete as it never works, it will be removed in a future release", false)]
+        IAsyncEnumerator<AppConfigurationKeyValueResource> IAsyncEnumerable<AppConfigurationKeyValueResource>.GetAsyncEnumerator(CancellationToken cancellationToken)
+        {
+            return GetAllAsync(cancellationToken: cancellationToken).GetAsyncEnumerator(cancellationToken);
```

Comment:
could we just yield break here to just return empty collection?
GetAllAsync method throws therefore this method throws.
This will cause exception to be thrown when they write:
```
await foreach (var item in collection) {}
```
we never want that.

Reply:
well yield break does not work here. Maybe we could write a private struct here to inherit `IAsyncEnumerator<AppConfigurationKeyValueResource>` and make sure its MoveNext always return false